code/modules/paperwork/paper.dm

```diff
@@ -163,6 +163,50 @@
 	else
 		return ..()
 
+/obj/item/paper/attack_animal(mob/living/simple_animal/M)
+	if(!isdog(M)) // Only dogs can eat homework.
+		return
+	var/mob/living/simple_animal/pet/dog/D = M
+	D.changeNext_move(CLICK_CD_MELEE)
+	if(world.time < D.last_eaten + 30 SECONDS)
+		to_chat(D, "<span class='warning'>You are too full to try eating [src] now.</span>")
+		return
+
+	D.visible_message("<span class='warning'>[D] starts chewing the corner of [src]!</span>",
+		"<span class='notice'>You start chewing the corner of [src].</span>",
+		"<span class='warning'>You hear a quiet gnawing, and the sound of paper rustling.</span>")
+	playsound(src, 'sound/effects/pageturn2.ogg', 100, TRUE)
+	if(!do_after(D, 10 SECONDS, FALSE, src))
+		return
+
+	if(world.time < D.last_eaten + 30 SECONDS) // Check again to prevent eating multiple papers at once.
+		to_chat(D, "<span class='warning'>You are too full to try eating [src] now.</span>")
+		return
+	D.last_eaten = world.time
+
+	// 90% chance of a crumpled paper with illegible text.
+	if(prob(90))
+		var/message_ending = "."
+		var/obj/item/paper/crumpled/P = new(loc)
+		P.name = name
+		if(info) // Something written on the paper.
+			var/new_text = strip_html_properly(info, MAX_PAPER_MESSAGE_LEN, TRUE) // Don't want HTML stuff getting gibberished.
+			P.info = Gibberish(new_text, 100)
+			message_ending = ", the drool making it an unreadable mess!"
```

Comment:
Put a hardcoded string here for the paper being unreadable without the text being turned into gibberish. If someone eats paper with a long enough message on it, it will crash the server. This is why paper being burnt doesn't turn the message to gibberish.

Reply:
It... how?
The `strip_html_properly()` proc is capping the length to 3072 characters maximum, and the `Gibberish()` is coming after that.
I tested it after running this code and it all worked fine, not even any lag spikes. (On a private test server)
```dm
/obj/item/paper/proc/aaaa()
	var/list/test = list()
	for(var/I in 1 to 100000)
		test += "[I]"
	info = jointext(test, "")
	return length(info) // 488895 characters
```
I'll remove it if it's an issue, but having tested it so much I can't see how it would be.

Reply:
Its happened before and took us months to pin down because it doesnt use hrefs. Remove it.